tap_postgres/tests/test_core.py

```diff
@@ -1,87 +1,163 @@
 """Tests standard tap features using the built-in SDK tests library."""
-
 import datetime
+import json
+import os
 
+import pendulum
+import pytest
+from faker import Faker
 from singer_sdk.testing import get_standard_tap_tests
+from sqlalchemy import Column, DateTime, Integer, MetaData, String, Table
 
 from tap_postgres.tap import TapPostgres
-from faker import Faker
-import pendulum
-import datetime
-import pytest
-import json
-from sqlalchemy import Table, Column, String, DateTime, Integer, MetaData
 
-SAMPLE_CONFIG = {
-    "start_date": pendulum.datetime(2022,11,1).to_iso8601_string(), 
-    "sqlalchemy_url": "postgresql://postgres:postgres@localhost:5432/postgres",
-}
 
+@pytest.fixture
+def sample_config_components(postgres_fixture):
+    """A config that uses the user/password/host/port settings to configure
+    database connection.
+    """
+    postgres_host_pytest_docker = postgres_fixture.get_addr("5432/tcp")[0]
+    return {
+        "start_date": pendulum.datetime(2022, 11, 1).to_iso8601_string(),
+        "host": os.getenv("TAP_POSTGRES_HOST", postgres_host_pytest_docker),
+        "port": os.getenv("TAP_POSTGRES_PORT", 5432),
+        "user": os.getenv("TAP_POSTGRES_USER", "postgres"),
+        "password": os.getenv("TAP_POSTGRES_PASSWORD", "mypass"),
+        "database": os.getenv("TAP_POSTGRES_DATABASE", "postgres"),
+    }
+
+
+@pytest.fixture
+def sample_config_sqlalchemy_url(postgres_fixture):
+    """A config that uses the sqlalchemy_url setting to configure
+    database connection.
+    """
+    postgres_host_pytest_docker = postgres_fixture.get_addr("5432/tcp")[0]
+    return {
+        "start_date": pendulum.datetime(2022, 11, 1).to_iso8601_string(),
+        "sqlalchemy_url": os.getenv(
+            "TAP_POSTGRES_SQLALCHEMY_URL",
+            (
+                "postgresql+psycopg2://postgres:mypass@"
```

Comment:
username/password/db should be config driven as well here I think similar to `sample_config_components`